lib/jellyfish_web/socket.ex

```diff
@@ -0,0 +1,75 @@
+defmodule JellyfishWeb.Socket do
+  @moduledoc false
+  @behaviour Phoenix.Socket.Transport
+
+  alias Jellyfish.{Room, RoomService}
+
+  @impl true
+  def child_spec(_opts) do
+    # We won't spawn any process, so let's return a dummy task
+    %{id: __MODULE__, start: {Task, :start_link, [fn -> :ok end]}, restart: :transient}
+  end
+
+  @impl true
+  def connect(state) do
+    with {:ok, peer_id} <- Map.fetch(state.params, "peer_id"),
+         {:ok, room_id} <- Map.fetch(state.params, "room_id"),
+         {:ok, room_pid} <- RoomService.find_room(room_id),
+         {:ok, :disconnected} <- Room.get_peer_connection_status(room_pid, peer_id) do
+      state =
+        state
+        |> Map.put(:room_pid, room_pid)
```

Comment:
If a room is registered, it should be posssible to use its id instead of pid. This way we are resiliant for room restarts - pid changes in such a case